vazhin/app.js

```diff
@@ -0,0 +1,283 @@
+            arrOfTasks.splice(indexOfInEditTask, 1, objOfTask);
+            inEditMode = false;
+            addBtn.innerHTML = "add task"
+        }
+        else {
+            arrOfTasks.push(objOfTask);
+        }
+    }
+
+    updateTasksInList();
+}
+
+function updateTasksInList() {
+
+    let previousTasks = document.querySelectorAll(".todo-list div");
+    for (let i = 0; i < previousTasks.length; i++) {
+        previousTasks[i].remove();
+    }
+
+    function compare(a, b) {
+        return a.priority - b.priority;
+    }
+
+    arrOfTasks.sort(compare);
+
+    for (let i = 0; i < arrOfTasks.length; i++) {
+
+        const todoDiv = document.createElement("div");
+        const task = document.createElement("li");
+        const priority = document.createElement("p");
+        const deadline = document.createElement("p");
+        const todoBtnsDiv = document.createElement("div");
+        const completedBtn = document.createElement('button');
+        const trashBtn = document.createElement('button');
+        const editBtn = document.createElement('button');
+
+        task.innerHTML = `task: ${arrOfTasks[i].task}`;
+        priority.innerHTML = `priority: ${arrOfTasks[i].priority}`;
+        deadline.innerHTML = `deadline: ${arrOfTasks[i].deadline}`;
+        completedBtn.innerHTML = '<i></i>';
+        trashBtn.innerHTML = '<i class="fas fa-trash-alt"></i>';
+        editBtn.innerHTML = 'edit';
+
+        noTasks.remove();
+        todoList.appendChild(todoDiv);
+        todoDiv.appendChild(task);
+        todoDiv.appendChild(priority);
+        todoDiv.appendChild(deadline);
+        todoDiv.appendChild(todoBtnsDiv);
+        todoBtnsDiv.appendChild(completedBtn);
+        todoBtnsDiv.appendChild(trashBtn);
+        todoBtnsDiv.appendChild(editBtn);
+
+        editBtn.classList.add("edit-btn");
+        todoBtnsDiv.classList.add("todo-btns-div");
+        completedBtn.classList.add("completed-btn");
+        trashBtn.classList.add("trash-btn");
+        todoDiv.classList.add("todo-div");
+        task.classList.add("display-5");
+        priority.classList.add("display-5");
+        deadline.classList.add("display-5");
+
+        taskText.value = "";
+        priorityInput.value = "";
+        deadlineInput.value = "";
+
+        if (arrOfTasks[i].priority === "1") {
+            priority.style.color = 'red';
+        }
+        else if (arrOfTasks[i].priority === "2") {
+            priority.style.color = 'blue';
+        }
+        else if (arrOfTasks[i].priority === "3") {
+            priority.style.color = 'green';
+        }
+
+        function checkDateOverdue() {
+
+            let thisTaskDeadline = Date.parse(arrOfTasks[i].deadline);
+            let today = Date.parse(changeDateFormat());
+            if (thisTaskDeadline <= today) {
+                deadline.classList.add("date-overdue");
+            }
+        }
+        checkDateOverdue();
+
+        trashBtn.addEventListener('click', function () {
+
```

Comment:
now, this is good, you have a reference to your button, but with my proposal above, you can bring back this reference using a querySelector for this button and the other buttons, or you can make this delete function global that takes an index and do add it in the getHTML function as well so that we don't create a new function for each item (while the function is doing the same thing)

if you go this way your getHTML function would probably need to take the index as well so that it attaches it as an id inside the template and be like:
`<button id="task-delete-${id}" onclick="deleteTodo">`... inside the template.
then deleteTodo would be:

    function deleteTodo(e) {
        const todoId = e.target.id.split("-").pop() //this will split task-delete-5 by the - and pop the 5 as string
        arrOfTasks.splice(parseInt(todoId), 1); // this will change the string into number then splice it from the array
    }

the same can be done for other function as well like the check item and the edit

_again all these comments are pro tips for optimization and performance tweaking, it doesn't mean your code is incorrect. But food for thought if it makes sense to you!_

Reply:
This is very interesting, I'll definitely think about it more to understand it fully. Thanks!